soc/arm/quicklogic_eos_s3/fpga_loader.c

```diff
@@ -0,0 +1,153 @@
+/*==========================================================
+*                                                          
+*   Copyright 2020 QuickLogic
+*   
+*   Licensed under the Apache License, Version 2.0 (the "License");
+*   you may not use this file except in compliance with the License.
+*   You may obtain a copy of the License at
+*   
+*       http://www.apache.org/licenses/LICENSE-2.0
+*   
+*   Unless required by applicable law or agreed to in writing, software
+*   distributed under the License is distributed on an "AS IS" BASIS,
+*   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+*   See the License for the specific language governing permissions and
+*   limitations under the License.
+*              
+*                                                          
+*    File   : fpga_loader.c
+*    Purpose: Contains functionality to load FPGA
+*                                                          
+*=========================================================*/
+
+#include <zephyr.h>
+#include <soc.h>
+#include <stdio.h>
+
+
+#define REG1                            (*(volatile uint32_t *)(0x40004610))
+#define REG2                            (*(volatile uint32_t *)(0x40004044))
+#define REG3                            (*(volatile uint32_t *)(0x4000404C))
+#define REG4                            (*(volatile uint32_t *)(0x40004064))
+
+#define REG5                            (*(volatile uint32_t *)(0x40004070))
+#define REG6                            (*(volatile uint32_t *)(0x4000411C))
+#define REG7                            (*(volatile uint32_t *)(0x40004054))
+#define REG8                            (*(volatile uint32_t *)(0x400047F8))
+
+#define REG9                            (*(volatile uint32_t *)(0x40014000))
+
+#define REG10                           (*(volatile uint32_t *)(0x400047F0))
+#define REG11                           (*(volatile uint32_t *)(0x400047F4))
+#define REG12                           (*(volatile uint32_t *)(0x40004088))
+#define REG13                           (*(volatile uint32_t *)(0x40004094))
+#define REG14                           (*(volatile uint32_t *)(0x400047F8))
+#define REG15                           (*(volatile uint32_t *)(0x40004040))
+#define REG16                           (*(volatile uint32_t *)(0x40004048))
+#define REG17                           (*(volatile uint32_t *)(0x4000404C))
+
+#define CFG_CTL_CFG_DATA                (*(volatile uint32_t *)(0x40014FFC))
+#define CFG_CTL_CFG_CTL                 (*(volatile uint32_t *)(0x40014000))
+
+static void enable_clocks()

```

Comment:
Could you compare your function to this one that is already implemented in `soc.c`? ([link](https://github.com/antmicro/zephyr/blob/eos-s3-support/soc/arm/quicklogic_eos_s3/soc.c#L115-L158)). I think we can use the function from `soc.c` here instead of writing a new one.